### PR TITLE
Image preprocessing block (flip, rotate, resize)

### PR DESCRIPTION
**Pull request by EmilyGavrilenko**
# Description

Add a block that supports resizing, flipping, and rotating

## Type of change
-   [X] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?
UI editor & with tests

## Any specific deployment considerations
n/a

## Docs
n/a

### CODE REVIEW

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        json_schema_extra={
+            "relevant_for": {
+                "task_type": {
+                    "values": ["rotate"],
+                    "required": True,
+                },
+            }
+        },
+    )
+    flip_type: Union[str, WorkflowParameterSelector(kind=[STRING_KIND]), Literal["vertical", "horizontal", "both"]] = Field(  # type: ignore
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:106]: let's remove str here

**Review by PawelPeczek-Roboflow — DISMISSED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            numpy_image=response_image,
+        )
+        return {"image": output_image}
+
+
+def apply_resize_image(
+    np_image: np.ndarray, width: Optional[int], height: Optional[int]
+) -> np.ndarray:
+    if width is None and height is None:
+        return np_image
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:175]:
```suggestion
        return np_image.copy()
```

We should return copy of the image to be consistent with remainder of this method where copy is returned

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        aspect_ratio = width / current_width
+        height = int(current_height * aspect_ratio)
+
+    resized_image = cv2.resize(np_image, (width, height), interpolation=cv2.INTER_AREA)
+    return resized_image
+
+
+def apply_rotate_image(np_image: np.ndarray, rotation_degrees: Optional[int]):
+    if rotation_degrees is None or rotation_degrees == 0:
+        return np_image
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:192]:
```suggestion
        return np_image.copy()
```

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    else:
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:193]: we already returned, else is not required (can save one indent level)

**Review by grzegorz-roboflow — CHANGES_REQUESTED** [review]

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py (hunk 2 above)
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:175]: 👍

File: inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py (hunk 3 above)
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py:192]: 👍

**Review by grzegorz-roboflow — APPROVED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**